pkg/app/reconciler.go

```diff
@@ -27,14 +28,21 @@ type Reconciler struct {
 	appClient       kcclient.Interface
 	log             logr.Logger
 	crdAppFactory   CRDAppFactory
+	deployFactory   deploy.Factory
 	appRefTracker   *reftracker.AppRefTracker
 	appUpdateStatus *reftracker.AppUpdateStatus
 }
 
 // NewReconciler constructs new Reconciler.
-func NewReconciler(appClient kcclient.Interface, log logr.Logger, crdAppFactory CRDAppFactory,
```

Comment:
with change to CRDAppFactory accepting deployFactory, this class would not have to change (since deployFactory is an implementation detail of how CRDAppFactory works).